futures-util/src/future/shared.rs

```diff
@@ -156,153 +203,76 @@ impl<F> Future for Shared<F>
                 }
             }
 
-            let _reset = Reset(&self.inner.notifier.state);
+            let _reset = Reset(&this.inner.notifier.state);
```

Comment:
Maybe call this `Bomb` like in `FuturesOnordered`

Reply:
`let _ = ...`

Reply:
`let _` doesn't do the same thing as `let _reset`-- `let _ = ` will instantly drop, rather than dropping at the end of the scope. Also, note that this isn't a `Bomb`-- it just resets the state, it doesn't trigger an abort.

Reply:
I didn't know that o_O I'm sure there has been a long debate about this once ^^'